docs/research/574-ls-projects-scout.md

```diff
@@ -0,0 +1,309 @@
+
+**Key Finding**: Projects in LangSmith are actually called "Sessions" at the API level (endpoint: `/sessions`). The Python SDK provides comprehensive project management through the sessions API, and langstar already references projects indirectly via the `--project` flag in the runs query command.
+
+## 1. Existing Langstar Code
+
+### Current Project References
+
+**CLI (`cli/src/commands/runs.rs:30-34`)**:
+```rust
+/// Project name or UUID to query runs from
+///
+/// Can be specified multiple times to query from multiple projects.
+#[arg(short, long = "project", value_name = "PROJECT")]
+pub projects: Vec<String>,
+```
+
+**SDK (`sdk/src/runs.rs:93-94`)**:
+```rust
+/// Session/project ID this run belongs to
+pub session_id: Uuid,
+```
+
+**Observation**:
+- Projects are currently only referenced as filtering criteria for runs
+- No direct project management commands exist (no `langstar project list`, `create`, `delete`, etc.)
+- The SDK uses `session_id` terminology matching the API
+- No existing `projects.rs` module in the SDK
+
+## 2. Python SDK Precedent
+
+**Location**: `reference/repo/langchain-ai/langsmith-sdk/code/python/langsmith/client.py`
+
+### Available Methods (Lines 3375-3780)
+
+1. **`list_projects()`** (line 3704)
+   - Filters: project_ids, name, name_contains, reference_dataset_id, reference_free, include_stats, metadata
+   - Returns: Iterator of `TracerSessionResult`
+   - Uses pagination with max 100 per page
+
+2. **`read_project()`** (line 3531)
+   - Parameters: project_id OR project_name, include_stats
+   - Returns: Single `TracerSessionResult`
+
+3. **`create_project()`** (line 3408)
+   - Parameters: project_name (required), description, metadata, upsert, reference_dataset_id
+   - Returns: `TracerSession`
+
+4. **`update_project()`** (line 3455)
+   - Parameters: project_id (required), name, description, metadata, end_time
+   - Note: name change only allowed if project has end_time (closed)
+   - Returns: `TracerSession`
+
+5. **`delete_project()`** (line 3780)
+   - Parameters: project_name OR project_id
+
+6. **`has_project()`** (line 3572)
+   - Parameters: project_name, optional project_id
+   - Returns: bool
+
+7. **`list_shared_projects()`** (line 3375)
+   - Not examined in detail
+
+### Schema: TracerSession (schemas.py:729-781)
+
+**Key Fields**:
+```python
+class TracerSession(BaseModel):
+    id: UUID
+    start_time: datetime
+    end_time: Optional[datetime]
+    description: Optional[str]
+    name: Optional[str]
+    extra: Optional[dict[str, Any]]  # Contains metadata and tags
+    tenant_id: UUID
+    reference_dataset_id: Optional[UUID]
+```
+
+**Important Comment** (line 732): *"Sessions are also referred to as 'Projects' in the UI."*
+
+## 3. API Endpoints
+
+**Base URL**: `https://api.smith.langchain.com`
+
+### REST Endpoints (from Python SDK)
```

Comment:
is there any ability to filter projects by name server side?

Reply:
Yes! The GET `/sessions` endpoint supports server-side filtering by name.

**Available filters** (from line 99 of the scout doc):
- `name` - exact name match
- `name_contains` - partial name match

This is also confirmed in Section 2 where Python SDK's `list_projects()` method (line 46) shows `name` and `name_contains` as filter parameters.